overlord/snapstate/handlers.go

```diff
@@ -1007,9 +1007,11 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 		logger.Noticef("cannot get model assertion: %v", model)
 		return
 	}
-	if model.Base() != "" && info.InstanceName() == "snapd" {
-		t.Logf("Requested daemon restart (snapd snap).")
-		st.RequestRestart(state.RestartDaemon)
+	if info.InstanceName() == "snapd" {
+		if err := canInstallSnapdSnap(st); err == nil {
+			t.Logf("Requested daemon restart (snapd snap).")
+			st.RequestRestart(state.RestartDaemon)
+		}
```

Comment:
else, should we log the error?

Reply:
In this case we just use canInstallSnapdSnap() as information - this indicates that maybe canInstallSnapdSnap needs to return a bool and a real error.